packages/context/test/unit/resolver.test.ts

```diff
@@ -169,3 +169,15 @@ describe('sync constructor & async property injection', () => {
     expect(t.bar).to.eql('BAR');
   });
 });
+
+describe('resolveValueOrPromise()', () => {
+  it('resolves both a promise and a value', async () => {
+    const value: String = 'hello';
+    const promise = new Promise<String>((resolve) => {
```

Comment:
This can be simplified too:

```ts
const promise = Promise.resolve(value);
```

No need to change this code, I am just trying to educate.